lightning/src/chain/onchaintx.rs

```diff
@@ -681,15 +732,32 @@ impl<ChannelSigner: Sign> OnchainTxHandler<ChannelSigner> {
 						//... we need to verify equality between transaction outpoints and claim request
 						// outpoints to know if transaction is the original claim or a bumped one issued
 						// by us.
-						let mut set_equality = true;
-						if request.outpoints().len() != tx.input.len() {
-							set_equality = false;
+						let mut are_sets_equal = true;
+						if !request.requires_external_funding() || !request.is_malleable() {
+							// If the claim does not require external funds to be allocated through
+							// additional inputs we can simply check the inputs in order as they
+							// cannot change under us.
+							if request.outpoints().len() != tx.input.len() {
```

Comment:
I'm sorry for saying this so late, and honestly, feel free to disregard this comment. But wouldn't it be more legible to just create two HashSets and then just check their equality?

Also, is there ever a scenario where someone could make either the inputs or the previous outputs contain the same value multiple times? My intuition is no, and it would obviously be invalid, but then a length sanity check with the sets would also make it more obvious.